src/lexer.l

```diff
@@ -183,19 +191,41 @@ struct|union|enum       {
                             struct_type = yytext;
                             return Parser::make_STRUCT(loc);
                         }
-<STRUCT,BRACE>{
-  "*"|")"|","           {
-                          if (YY_START == STRUCT)
-                          {
-                            // Finished parsing the typename of a cast or a call arg
-                            // Put the cast type into a canonical form by trimming
-                            // and then inserting a single space.
-                            yy_pop_state(yyscanner);
-                            unput(yytext[0]);
-                            return Parser::make_IDENT(struct_type + " " + trim(buffer), loc);
-                          }
-                          buffer += yytext[0];
+<AFTER_COLON>{
+  {hspace}+             { loc.step(); }
+  {vspace}+             { loc.lines(yyleng); loc.step(); }
+  struct|union|enum     {
+                          yy_pop_state(yyscanner);
+                          yy_push_state(STRUCT_AFTER_COLON, yyscanner);
+                          buffer.clear();
+                          struct_type = yytext;
+                          return Parser::make_STRUCT(loc);
+                        }
+  .                     { unput(yytext[0]); yy_pop_state(yyscanner); }
+}
+<STRUCT_AFTER_COLON>{
+  "{"                   {
+                          yy_pop_state(yyscanner);
+                          unput(yytext[0]);
+                          return Parser::make_IDENT(struct_type + " " + trim(buffer), loc);
                         }
+  .                     { buffer += yytext[0]; }
+  \n                    buffer += '\n'; loc.lines(1); loc.step();
+}
+<STRUCT,BRACE>{
+  "*"|")"|","|{space}+"$" {
```

Comment:
Why are the space changes needed in lexer.l (this line and also in the `AFTER_COLON` section)? Why is space handled differently in this section?

Reply:
`AFTER_COLON` is an `%x` section, IIUC the usual rules do not apply there. Here the space handling should be a part of the logic ensuring `struct test $x` (for example) is parsed correctly.

I can take a closer look at it if that did not make much sense, that's what I remember from writing this code.